java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TestFinder.java

```diff
@@ -0,0 +1,235 @@
+                    for (TestMethodController.TestMethod testMethod : testMethods) {
+                        if (testClassName == null) {
+                            testClassName = testMethod.getTestClassName();
+                        }
+                        if (testClassLine == null) {
+                            testClassLine = testMethod.getTestClassPosition() != null
+                                    ? Utils.createPosition(fo, testMethod.getTestClassPosition().getOffset()).getLine()
+                                    : null;
+                        }
+                        String id = testMethod.getTestClassName() + ':' + testMethod.method().getMethodName();
+                        String fullName = testMethod.getTestClassName() + '.' + testMethod.method().getMethodName();
+                        int testLine = Utils.createPosition(parserResult.getSnapshot().getSource().getFileObject(), testMethod.start().getOffset()).getLine();
+                        tests.add(new TestSuiteInfo.TestCaseInfo(id, testMethod.method().getMethodName(), fullName, url, testLine, TestSuiteInfo.State.Loaded, null));
+                    }
+                    client.notifyTestProgress(new TestProgressParams(url, new TestSuiteInfo(testClassName, url, testClassLine, TestSuiteInfo.State.Loaded, tests)));
+                }
+            }
+        }
+    }
+
+    private void store(FileObject indexFolder, URL url, String resourceName, List<TestMethodController.TestMethod> methods) {
+        try {
+            File cacheRoot = FileUtil.toFile(indexFolder);
+            File output = new File(cacheRoot, resourceName + ".tests"); //NOI18N
+            output.getParentFile().mkdirs();
+            final PrintWriter pw = new PrintWriter(new OutputStreamWriter(new FileOutputStream(output), "UTF-8")); //NOI18N
+            boolean printHeader = true;
+            try {
+                for (TestMethodController.TestMethod method : methods) {
+                    if (printHeader) {
+                        pw.print("url: "); //NOI18N
+                        pw.println(url.toString());
+                        pw.print("class: "); //NOI18N
+                        pw.print(method.getTestClassName());
+                        pw.print(':'); //NOI18N
+                        pw.println(method.getTestClassPosition().getOffset());
+                        printHeader = false;
+                    }
+                    pw.print("method: "); //NOI18N
+                    pw.print(method.method().getMethodName());
+                    pw.print(':'); //NOI18N
+                    pw.println(method.start().getOffset());
+                }
+            } finally {
+                pw.close();
+            }
+        } catch (IOException ex) {
+        }
+    }
+
+    private static FileObject getCacheRoot(URL root) throws IOException {
+        final FileObject dataFolder = CacheFolder.getDataFolder(root, true);
```

Comment:
I guess this somewhat duplicates the logic that creates `Context.getIndexFolder()`; couldn't there be some API in `parsing.indexing` for this ?

Reply:
A set of clients calling `CacheFolder.getDataFolder()` is small. I would prefer to keep indexing API as it is for now.